src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -388,13 +377,33 @@ private void handleSiteUpdate(RoutingContext rc) {
                     ResponseUtil.error(rc, 400, "Invalid parameter for visible: " + visibleParam);
                 }
             }
+            if (name != null) {
+                if (validateSiteName(rc, name)) return;
+                existingSite.setName(name);
+            }
 
             uploadSiteToStoreWriterAndWriteExistingSiteToResponse(existingSite, rc);
         } catch (Exception e) {
             rc.fail(500, e);
         }
     }
 
+    private boolean validateSiteName(RoutingContext rc, String name) {
```

Comment:
This feels like the wrong way round to me - I would expect something called validateSiteName to return true if is was valid, and false is it was not.